### DISCUSSION TITLE
Add backup/restore endpoints/services

### DISCUSSION THREAD

**Discussion by rijkvanzanten** [directus/directus#5107] [Feature Requests]
Will make a SQL dump, and make a copy of all files, and saves them on disk as a zip in the configured backup location. Restore resets the DB back to the dump, and overwrites the file locations with the backed-up files

**Comment by mscbpi** [comment]
@rijkvanzanten  would the SQL dump depends on the SQL engine it's run on? 
Do you think an export in an agnostic format could be possible ? 

Getting the data through the API and be able to feed a new install, who might have a different DB engine.

**Comment by rijkvanzanten** [comment]
That's what we call import / export, not backup / restore 😄 

Exporting data can already be done to a certain extend by adding `?export=json` or  `?export=csv` to any endpoint 👍

**Comment by mscbpi** [comment]
thanks 👍

**Comment by benhaynes** [comment]
While _restore_ is really useful for migrations, _backup_ is important for automating backups on Cloud, etc.

**Comment by gnimmelf** [comment]
Hi, guys,

just a though (again). 

Having some easy to follow dev/stage-to-prod-how-tos in the documentation would be super nice. - It may be "outside" Directus' scope, but I, as many others I would guess, pick up Directus primarily due to the easiness it puts into maintaining a backend so I can focus on the front / jamstack.

For my own case, there is a whole new ecosystem I have had to stick my head into, postgres, docker/-compose, svelte/-kit, Cloudflare Pages and Directus, and the deployment pipeline is by far the most daunting problem for me as primarily a front-end guy without access to backend dev resources.

I guess you's already aware, and have loads of work, but I'm stressing this because it  sort of is crucial; setting up a local dev  jamstack is a piece of cake, setting up Directus and creating within it is a piece of cake ('cause it's awesome), but that last part, pipelining the local Directus dev DB to prod, requires a lot of research if not already familiar with DB-maintenance, which I would guess is totally on the other end of what loads of early Directus v9 adapters are familiar with.

Cheers!
Flemming

**Reply by tylerforesthauser** [reply]
💯

**Reply by benhaynes** [reply]
We're actually working on the schema backup/restore endpoints now! Docs will come with that release.

**Reply by florianjs** [reply]
@gnimmelf Exactly the point where I am today. Struggling the whole day trying to dump a simple postgresql database from DO ... My brain hurts.

**Reply by NilsBaumgartner1994** [reply]
It would be very nice to add a nice and easy way to import / export a whole database via a menu in directus. This would make schedule work for backups very nice.
Also a kind of backup policy would be also very nice, like a rule "Never Export the field phone number of table contact"

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
